src/crypto/CHIPCryptoPAL.h

```diff
@@ -320,6 +347,47 @@ class P256Keypair : public ECPKeypair<P256PublicKey, P256ECDHDerivedSecret, P256
     void Clear();
 };
 
+/**
+ * @brief Convert a raw ECDSA signature to ASN.1 signature (per X9.62) as used by TLS libraries.
+ *
+ * Errors are:
+ *   - CHIP_ERROR_INVALID_ARGUMENT on any argument being invalid (e.g. nullptr), wrong sizes,
+ *     wrong or unsupported format,
+ *   - CHIP_ERROR_BUFFER_TOO_SMALL on running out of space at runtime.
+ *   - CHIP_ERROR_INTERNAL on any unexpected processing error.
+ *
+ * @param[in] fe_length_bytes Field Element length in bytes (e.g. 32 for P256 curve)
+ * @param[in] raw_sig Raw signature of <r,s> concatenated
+ * @param[in] raw_sig_length Raw signature length (MUST be 2*`fe_length_bytes` long)
+ * @param[out] out_asn1_sig ASN.1 DER signature format output buffer
+ * @param[in] out_asn1_sig_length ASN.1 DER signature format output buffer length. Must have space for at least
+ * kMax_ECDSA_X9Dot62_Asn1_Overhead.
+ * @param[out] out_asn1_sig_actual_length Final computed size of signature.
+ * @return Returns a CHIP_ERROR on error, CHIP_NO_ERROR otherwise
+ */
+CHIP_ERROR EcdsaRawSignatureToAsn1(size_t fe_length_bytes, const uint8_t * raw_sig, size_t raw_sig_length, uint8_t * out_asn1_sig,
```

Comment:
would we be able to use spans to represent `buffer + length` pairs?

Reply:
Will do as a follow-up, since it's a lot of utilitarian surgery and this PR is very important to land.

Reply:
I will also note that all other APIs in CHIPCryptoPAL.h use `buf + length` pairs

Reply:
I requested this before and got the same answer :D but at that time I also had "our current span/buffer does not support  const" but we fixed that in the mean time.

Nitpicky though to ask for "please change this new code, even if old code is already using this syntax" so I agree a separate PR may be better.